tests/requirements.txt

```diff
@@ -5,8 +5,8 @@ pytest-xdist==1.28.0
 pytest>=4.4,<4.5
 notario>=0.0.13
 ansible>=2.8,<2.9
-netaddr
+netaddr # For the ansible filter: ips_in_ranges
```

Comment:
I don't think you need to update this file.

Reply:
Without this the CI won't install netaddr. Because a grep is used here:
https://github.com/ceph/ceph-build/blob/08b569d032e8d67643fdbbb33fb6006b3f8d635e/ceph-ansible-pr-syntax-check/build/build#L11

Adding a comment containing the "ansible" string is more like a workaround to ensure that CI job passes. (Adding the "ansible" string include that line in the grep result and thus the netaddr requirement is installed.)

Reply:
No, we install the deps via tox https://github.com/ceph/ceph-ansible/blob/master/tox.ini#L418 otherwise the ipaddr filter would have never work.

Reply:
Not for the ceph-ansible-pr-syntax-check job, it installs deps via tox like this:
`"$VENV"/pip install -r <(grep ansible "$WORKSPACE"/ceph-ansible/tests/requirements.txt)`

Effectively passing only *tests/requirements.txt#L7* containing `ansible>=2.8,<2.9`.

The *ceph-ansible-pr-syntax-check* never exercice any code that use the ipaddr filter, and since ansible does conditional loading it will work until the filter is actually used.

I do however agree that me adding that comment in requirements.txt is a workaround. I will remove it and then we can solve the problem in ceph/ceph-build instead.

Reply:
Here is the CI log showing that dependencies are not properly installed:
https://2.jenkins.ceph.com/job/ceph-ansible-pr-syntax-check/1204/console

Reply:
I pushed this PR https://github.com/ceph/ceph-build/pull/1391 which I belive should fix the problem with netaddr not being pulled in as a dependency in the *ceph-ansible-pr-syntax-check* job.